shopinvader_api_address/models/res_partner.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import MissingError, UserError
+
+from ..schemas import AddressInput, AddressSearch,AddressUpdate
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    def _billing_address_already_used(self):
+        self.ensure_one()
+        move_id = self.env["account.move"].search([("commercial_partner_id","=",self.id)],limit=1)
+        return len(move_id)>0
+
+    @api.model
+    def _create_vals_shopinvader_address(self, data: AddressInput,authenticated_partner_id:"ResPartner",type:str) -> dict:
+
+        vals = {
+            "name": data.name or "",
+            "street": data.street or "",
+            "street2": data.street2 or "",
+            "zip": data.zip or "",
+            "city": data.city or "",
+            "phone": data.phone or "",
+            "email": data.email or "",
+            "state_id": data.state or None,
+            "country_id": data.country or None,
+            "parent_id": authenticated_partner_id.id,
+            "type": type or "",
+        }
+        return vals
```

Comment:
I think this mapping between the schemas and the odoo models should be done in `routers`. The domain models should not have knowledge about the API schemas.

We should not put domain logic/knowledge in the service layer, but mapping the input/output schemas is definitely the job of the services layer.

Reply:
Now, remembering again that we are in Odoo and someone will want to override that :)
But still. I think this mapping code should not be in `models`. Maybe in some `AbstractModel`s in `routers`?

Reply:
The AbstraModel seems a great trade off

Reply:
cc @lmignon food for thought in the bus :)

Reply:
I tried to extract the mapping using an AbtractModel : https://github.com/shopinvader/odoo-shopinvader/pull/1361/files#diff-cd672fd3ed425e60178156fc9e15fc26f1e683346017bd0f1d1f258938a3472bR92

Let me know your thoughts

PS: not sure about the name of the class